internal/config/cli.go

```diff
@@ -44,6 +65,9 @@ type ContainerCLIConfig struct {
 	// is required, the features.allow-ldconfig-from-container feature gate must
 	// be enabled explicitly.
 	Ldconfig ldconfigPath `toml:"ldconfig"`
+	// CUDACompatMode sets the mode to be used to make CUDA Forward Compat
+	// libraries discoverable in the container.
+	CUDACompatMode cudaCompatMode `toml:"cuda-compat-mode,omitempty"`
```

Comment:
Should this rather be at:
```
nvidia-container-runtime.cuda-compat-mode = ldconfig
```
or
```
nvidia-container-runtime.modes.legacy.cuda-compat-mode = ldconfig
```

I think the latter since it actually only affects the behaviour of the `legacy` runtime mode, with the hook always being injected in other modes except if it is explicitly opted out of by the feature flag.

Reply:
I don't have a strong opinion here, but I think your latter suggestion makes sense. At first, I thought this made more sense in the `nvidia-container-cli` section, but because of the value `hook`, which only applies when the nvidia-contianer-runtime is used, I think it is okay to nest this option under `nvidia-container-runtime.modes.legacy`

Reply:
Yes, I think it's cleaner this way. I have updated the implementation accordingly.